paymentsheet/src/test/java/com/stripe/android/paymentelement/confirmation/gpay/GooglePayConfirmationDefinitionTest.kt

```diff
@@ -0,0 +1,482 @@
+                        currency = "USD",
+                    ),
+                ),
+            ),
+            shouldHaveCurrencyCodeFailure = false,
+            merchantCurrencyCode = null,
+        )
+
+    @Test
+    fun `'Launch' action should be returned if currency code is provided with deferred intent in payment mode`() =
+        runActionTest(
+            initializationMode = PaymentElementLoader.InitializationMode.DeferredIntent(
+                intentConfiguration = PaymentSheet.IntentConfiguration(
+                    mode = PaymentSheet.IntentConfiguration.Mode.Payment(
+                        amount = 1099,
+                        currency = "USD",
+                    ),
+                ),
+            ),
+            shouldHaveCurrencyCodeFailure = false,
+            merchantCurrencyCode = "USD",
+        )
+
+    @Test
+    fun `On 'launch', should create google pay launcher properly`() = runTest {
+        RecordingGooglePayPaymentMethodLauncherFactory.test(mock()) {
+            val definition = createGooglePayConfirmationDefinition(factory)
+            val launcher = FakeActivityResultLauncher<GooglePayPaymentMethodLauncherContractV2.Args>()
+
+            definition.launch(
+                confirmationOption = GOOGLE_PAY_CONFIRMATION_OPTION,
+                intent = PAYMENT_INTENT,
+                arguments = Unit,
+                launcher = launcher,
+            )
+
+            val createGooglePayLauncherCall = createGooglePayPaymentMethodLauncherCalls.awaitItem()
+
+            assertThat(createGooglePayLauncherCall.activityResultLauncher).isEqualTo(launcher)
+            assertThat(createGooglePayLauncherCall.skipReadyCheck).isTrue()
+            assertThat(createGooglePayLauncherCall.cardBrandFilter).isEqualTo(DefaultCardBrandFilter)
+
+            assertThat(createGooglePayLauncherCall.config.environment).isEqualTo(GooglePayEnvironment.Test)
+            assertThat(createGooglePayLauncherCall.config.merchantName).isEqualTo("Test merchant Inc.")
+            assertThat(createGooglePayLauncherCall.config.merchantCountryCode).isEqualTo("US")
+            assertThat(createGooglePayLauncherCall.config.allowCreditCards).isTrue()
+            assertThat(createGooglePayLauncherCall.config.existingPaymentMethodRequired).isTrue()
+            assertThat(createGooglePayLauncherCall.config.isEmailRequired).isFalse()
+            assertThat(createGooglePayLauncherCall.config.billingAddressConfig.isRequired).isTrue()
+            assertThat(createGooglePayLauncherCall.config.billingAddressConfig.isPhoneNumberRequired).isFalse()
+            assertThat(createGooglePayLauncherCall.config.billingAddressConfig.format)
+                .isEqualTo(GooglePayPaymentMethodLauncher.BillingAddressConfig.Format.Full)
+        }
+    }
+
+    @Test
+    fun `On 'launch', should use payment intent currency code if available`() = runTest {
+        val googlePayLauncher = mock<GooglePayPaymentMethodLauncher>()
```

Comment:
There are a few places where I am using `GooglePayPaymentMethodLauncher` as a mock. Unfortunately, `GooglePayPaymentMethodLauncher` is a public API launcher class with no interface so we cannot update it to be mockable.

I thought about making a wrapper class with an interface for it but this seems like an extra unnecessary layer that is only beneficial for testing. Wondering what others think on this.